### PR TITLE
Add permissions API and move all feature content and js scripts into feature-page component

### PR DESCRIPTION
**Pull request by kevinshen56714**
In this PR, all content and js in the `feature.html` are moved into the `chromedash-feature-page` component. To achieve that, several changes are made:

1. `permission_api.py` and `permission_api_test.py` are added, and the corresponding /currentuser/permission route is added.
2. `static/js-src/feature-page.js` is safely removed. All the js functions and eventlisteners are moved to the `chromedash-feature-page` component with some modification to utilize the state management and event decorators of Lit.
3. The tests in `chromedash-feature-page_test.js` are improved for better coverage, and the original feature page template tests in `featuredetail_test.py` are reduced.
4. The content of `feature.html` is now a single Lit element!

### CODE REVIEW

**Comment by kevinshen56714** [comment]
@jrobbins This PR is ready for review as well.

Also, I think it would be a good time to deploy a staging server version as this basically wraps up the feature page transition for now (will come back later when we set up the router and stuff).

I will start to work on another page (probably the myfeatures or features page).

Thank you!!

**Review by jrobbins — APPROVED** [review]
Great work.  Please consider the comments and then make changes that you think are appropriate.

File: templates/feature.html [hunk 1]
```diff
@@ -9,99 +9,14 @@
...
-        {% if feature.browsers.chrome.status.text == "Removed" %} (removed){% endif %}
-  </h2>
-
-</div>
-{% endblock %}
-
-
 {% block content %}
-  <!-- TODO(kevinshen56714): get feature_id from SPA router -->
+  <!-- TODO(kevinshen56714): get feature_id and context_link from SPA router -->
```
[the remark this answers is not in the document]
Comment (kevinshen56714) [reply templates/feature.html:17]: Fixed it! Seeing the comment in the source is indeed kind of weird.

File: static/elements/chromedash-feature-page.js [hunk 2]
```diff
@@ -267,17 +385,29 @@ export class ChromedashFeaturePage extends LitElement {
...
+      <link rel="stylesheet" href="/static/css/forms.css">
+      <link rel="stylesheet" href="/static/css/guide.css">
       ${this.loading ?
         html`
           <div class="loading">
             <div id="spinner"><img src="/static/img/ring.svg"></div>
           </div>` :
         html`
+          ${this.renderSubHeader()}
           <div id="feature">
```
[the remark this answers is not in the document]
Comment (kevinshen56714) [reply static/elements/chromedash-feature-page.js:411]: Added a TODO comment for this.

File: templates/feature.html [hunk 3]
```diff
@@ -9,99 +9,14 @@
...
   <chromedash-feature-page
```
[the remark this answers is not in the document]
Comment (kevinshen56714) [reply templates/feature.html:18]: Yay!

File: static/elements/chromedash-feature-page.js [hunk 4]
```diff
@@ -58,49 +59,166 @@ export class ChromedashFeaturePage extends LitElement {
...
+          `: nothing}
+        </div>
+        <h2 id="breadcrumbs">
+          <a href="${this.contextLink}">
+            <iron-icon icon="chromestatus:arrow-back"></iron-icon>
+          </a>
+          <a href="/feature/${this.featureId}">
+            Feature: ${this.feature.name}
+          </a>
+          ${choose(this.feature.browsers.chrome.status.text,
```
[the remark this answers is not in the document]
Comment (kevinshen56714) [reply static/elements/chromedash-feature-page.js:211]: Changed it to just render the text directly. I didn't notice the content is almost exact same as what API returns.

File: api/permissions_api_test.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  def tearDown(self):
+    testing_config.sign_out()
+    ramcache.flush_all()
+    ramcache.check_for_distributed_invalidation()
+
+  def test_get__anon(self):
+    """Returns no user object if not signed in"""
+    testing_config.sign_out()
+    with test_app.test_request_context(self.request_path):
+      actual_response = self.handler.do_get()
```
[the remark this answers is not in the document]
Comment (kevinshen56714) [reply api/permissions_api_test.py]: Changed all to `actual`

**Pull request merged**